### ISSUE TITLE
Generated getters and setters don't match

### ISSUE THREAD

**Issue by jpnurmi** [canonical/dbus.dart#97]
Generated read-write remote-object properties are broken. Async getters returning Future<T> are not compatible with setters, that cannot be async, by the same name.

```bash
$ dart-dbus generate-remote-object org.freedesktop.NetworkManager.Device.xml # [1]
```
=>
```dart
  /// Gets org.freedesktop.NetworkManager.Device.Managed
  Future<bool> get Managed async {
    var value = await getProperty('org.freedesktop.NetworkManager.Device', 'Managed');
    return (value as DBusBoolean).value;
  }

  /// Sets org.freedesktop.NetworkManager.Device.Managed
  set Managed (bool value) {
    setProperty('org.freedesktop.NetworkManager.Device', 'Managed', DBusBoolean(value));
  }
```
=>
> The return type of getter 'Managed' is 'Future<bool>' which isn't assignable to the type 'bool' of its setter 'Managed'.  Try changing the types so that they are compatible.

[1] https://cgit.freedesktop.org/NetworkManager/NetworkManager/plain/introspection/org.freedesktop.NetworkManager.Device.xml

**Comment by robert-ancell** [comment]
I had been considering replacing the getters and setters with just methods, e.g. `getManaged()`, `setManaged()`, `callSomeFunction()`. It's not quite as nice but perhaps a bit clearer that these are async and are actually a method call underneath.

Where I've used the generated code I tend to modify it to be more appropriate to the service you are implementing. e.g. in https://github.com/canonical/networkmanager.dart I cache the properties then use getters/setters.

**Comment by robert-ancell** [comment]
I think perhaps the best strategy with the code generator is to generate *reliable* and *predictable* code, and leave it to the developer to decide if they want to make it have better usability.

**Comment by jpnurmi** [comment]
> Where I've used the generated code I tend to modify it to be more appropriate to the service you are implementing. e.g. in https://github.com/canonical/networkmanager.dart I cache the properties then use getters/setters.
>
> I think perhaps the best strategy with the code generator is to generate reliable and predictable code, and leave it to the developer to decide if they want to make it have better usability.

Agreed. In the beginning I was trying hard not to touch the generated code, but I've noticed the temptation to modify the generated code to optimize it for my specific use cases. It could be a matter of "branding" it as a generator that just creates a super convenient starting point.

Oh, and as you might have guessed based on the bug reports, I had missed your networkmanager package. Good stuff :)

**Comment by robert-ancell** [comment]
Should be fixed in 0.0.0-dev.20, please re-open if not. Thanks!
